app/javascript/src/ChangePassword/components/ChangePassword.jsx

```diff
@@ -0,0 +1,105 @@
+// @flow
+
+import React, { useState } from 'react'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+import { LoginPage, Form, ActionGroup, Button } from '@patternfly/react-core'
+import { HiddenInputs, FlashMessages } from 'LoginPage'
+import { PasswordInput } from 'ChangePassword'
+import type { FlashMessage } from 'Types'
+import { validateForm } from 'LoginPage/utils/formValidation'
+
+import 'LoginPage/assets/styles/loginPage.scss'
+import brandImg from 'LoginPage/assets/images/3scale_Logo_Reverse.png'
+import PF4DownstreamBG from 'LoginPage/assets/images/PF4DownstreamBG.svg'
+
+type Props = {
+  lostPasswordToken: ?string,
+  url: string,
+  errors: (?FlashMessage)[]
+}
+
+const validationConstraints = {
+  'user[password]': {
+    presence: true,
+    length: { minimum: 1 }
+  },
+  'user[password_confirmation]': {
+    presence: true,
+    length: { minimum: 1 },
+    equality: 'user[password]'
+  }
+}
+
+const ChangePassword = ({ lostPasswordToken, url, errors }: Props) => {
+  const [isFormDisabled, setIsFormDisabled] = useState(true)
+  const [passwordsMatch, setPasswordsMatch] = useState(true)
+
+  const onChange = (event) => {
+    const errors = validateForm(event.currentTarget, validationConstraints)
+
+    if (errors) {
+      const passwordConfirmationError = errors['user[password_confirmation]'][0]
+      const doesPasswordsMatch = !passwordConfirmationError.includes('is not equal to')
+      setPasswordsMatch(doesPasswordsMatch)
+      setIsFormDisabled(true)
+    } else {
+      setPasswordsMatch(true)
+      setIsFormDisabled(false)
+    }
+  }
+
+  return (
+    <LoginPage
+      brandImgSrc={brandImg}
+      brandImgAlt='Red Hat 3scale API Management'
+      backgroundImgSrc={PF4DownstreamBG}
+      backgroundImgAlt='Red Hat 3scale API Management'
+      loginTitle='Change Password'
+      footer={null}
+    >
+      {errors && <FlashMessages flashMessages={errors}/>}
+      <Form
+        action={url}
+        noValidate
+        id='edit_user_2'
+        acceptCharset='UTF-8'
+        method='post'
+        onChange={onChange}
+      >
+        <input type='hidden' name='_method' value='put' />
+        <HiddenInputs />
```

Comment:
Why not passing here `isPasswordReset`? Is that flag for anything else?

Reply:
That flag is need in the Password Reset form, rendered in Login Page, but not here

Reply:
OK, although it's a bit confusing since this component has the very same name of the flag.